client/src/daemon_launcher.cpp

```diff
@@ -38,7 +44,51 @@ pid_t daemon_launcher::start() {
     return ret;
 }
 
-#ifndef __APPLE__
+#include <iostream>
+
+#ifdef __APPLE__
+void daemon_launcher::open(simpleipc::client::service_client_impl& impl) {
+    remove(service_path.c_str());
+
+    int kq = kqueue();
+    int f = ::open(FileUtil::getParent(service_path).c_str(), O_RDONLY);
+
+    pid_t proc = start();
+
+    struct kevent k_ev;
+    EV_SET(&k_ev, f, EVFILT_VNODE, EV_ADD | EV_CLEAR, NOTE_WRITE, 0, NULL);
+    kevent(kq, &k_ev, 1, NULL, 0, NULL);
+
+    EV_SET(&k_ev, proc, EVFILT_PROC, EV_ADD | EV_CLEAR, NOTE_EXIT, 0, NULL);
+    kevent(kq, &k_ev, 1, NULL, 0, NULL);
+
+    struct kevent ev_set;
+    struct kevent ev_list[2];
+    int n;
+
+    struct timespec timeout;
+    timeout.tv_sec = 10;
+    timeout.tv_nsec = 0;
+
+    n = kevent(kq, NULL, 0, ev_list, 2, &timeout);
```

Comment:
I wonder if it's enough to specify timeout only here? Is it not possible for kevent to get interrupted before in some cases?